packages/design-system/src/components/dropDown/index.js

```diff
@@ -85,12 +87,34 @@ export const DropDown = forwardRef(
     ref
   ) => {
     const internalRef = useRef();
+    const placementRef = useRef(placement);
+    const [dynamicPlacement, setDynamicPlacement] = useState(placement);
 
     const { activeOption, isOpen, normalizedOptions } = useDropDown({
       options,
       selectedValue,
     });
 
+    const positionPlacement = useCallback(
+      (popupDimensions) => {
+        // check to see if there's an overlap with the window edge
+        const neededVerticalSpace = popupDimensions?.offset.bottom;
+        // if the popup was assigned as bottom we want to always check it
+        if (
+          popupDimensions &&
+          dynamicPlacement.startsWith('bottom') &&
+          neededVerticalSpace >= window.innerHeight
+        ) {
+          setDynamicPlacement(PLACEMENT.TOP);
+        }
+      },
+      [dynamicPlacement]
+    );
```

Comment:
This is essentially the exact same callback as used in tooltip, should I move into a shared util file?

Reply:
I think it's okay keeping it out of a util for now. If it happens again then let's do it!